### PR TITLE
DATAUP-210 File size error dropzone

### PR DESCRIPTION
**Pull request by eamahanna**
# Description of PR purpose/changes

This PR adds a file too large message as detailed in DATAUP-210. 
Screen Shot 2020-10-27 at 6 03 13 PM

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-210

- [ ] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
* execute `make test-frontend-unit`
* Navigate to the data import tab on localhost. Add a file that exceeds 20 gb. You will now be able to see the error and click on the link to navigate to the globus docs or a globus account, depending on how you are signed in.

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the travis build passes)

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js [hunk 1]
```diff
@@ -122,32 +126,41 @@ define([
...
+                    $errorElem.find('#error_icon').css('display', 'flex');
+                    this.removeProgressBar($dropzoneElem);
+
+                    // Set error message
                     let errorText = 'unable to upload file!';
-                    if (err && err.xhr && err.xhr.responseText) {
-                        errorText = err.xhr.responseText;
+                    var $errorMessage = $errorElem.find('#error_message');
+
+                    // I don't know how to determine if the file was too big other than looking at the preview message
```
Comment (leia-sefkin) [inline kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:143]: yeah looks like dropzone doesn't offer much help here, there's a maxfilesexceeded event but not for maxFilesize 
https://www.dropzonejs.com/#events 

this approach seems to make sense given that

File: kbase-extension/static/kbase/templates/data_staging/dropped_file.html [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
-            <div class="progress-bar progress-bar-success" style="width:0%;" data-dz-uploadprogress></div>
+<div class="row file-row dz-file">
+    <div id="dz_file_row_1" class="row">
+        <div class="col-md-7">
+            <div class="dz-file__name" data-dz-name></div>
+        </div>
+        <div class="col-md-1 ">
+            <span class="size pull-right" data-dz-size></span>
+        </div>
+        <div id="upload_progress_and_cancel" style="display: inline;">
```
Comment (leia-sefkin) [inline kbase-extension/static/kbase/templates/data_staging/dropped_file.html:9]: any way to remove this inline style?

**Review by leia-sefkin — APPROVED** [review]
A couple of places I could see minor improvements, but looks good overall, and works when I tested locally

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js (hunk 1 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:127]: The `erroredFile` is a standard WebAPI [File](https://developer.mozilla.org/en-US/docs/Web/API/File) object, so it has a size attribute which gives up the size in bytes. In theory, that should be the error triggered by the `maxFilesize` option way up above where the Dropzone is created. Maybe that's useful?

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js [hunk 3]
```diff
@@ -168,6 +186,37 @@ define([
             $('#clear-all-btn').remove();
         },
 
+        makeGlobusErrorLink: function(globusUrlLinked) {
+            let url = 'https://docs.kbase.us/data/globus';
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:190]: Should this be part of the `uploadConfig` object?

Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:190]: Also, this and `$globusErrorLink` should be `const`

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js [hunk 4]
```diff
@@ -168,6 +186,37 @@ define([
...
+
+            let $globusErrorLink = $("<a>")
+            .attr('id', 'globus_error_link')
+            .attr('href', url)
+            // .attr('target', '_blank')
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:195]: can delete this comment

**Review by briehl — APPROVED** [review]
Just a couple minor suggestions. But looks good!

**Pull request merged**